bazel/cc_grpc_library.bzl

```diff
@@ -2,70 +2,94 @@
 
 load("//bazel:generate_cc.bzl", "generate_cc")
 
-def cc_grpc_library(name, srcs, deps, proto_only, well_known_protos, generate_mocks = False, use_external = False, **kwargs):
-  """Generates C++ grpc classes from a .proto file.
+def cc_grpc_library(
+        name,
+        srcs,
+        deps,
+        proto_only = False,
+        well_known_protos = True,
+        generate_mocks = False,
+        use_external = False,
+        grpc_only = False,
+        **kwargs):
+    """Generates C++ grpc classes for services defined in a proto file.
 
-  Assumes the generated classes will be used in cc_api_version = 2.
+    If grpc_only is True, this rule is compatible with proto_library and
+    cc_proto_library native rules such that it expects proto_library target
+    as srcs argument and generates only grpc library classes, expecting
+    protobuf messages classes library (cc_proto_library target) to be passed in
+    deps argument. By default grpc_only is False which makes this rule to behave
+    in a backwards-compatible mode (trying to generate both proto and grpc
+    classes).
 
-  Arguments:
-      name: name of rule.
-      srcs: a single proto_library, which wraps the .proto files with services.
-      deps: a list of C++ proto_library (or cc_proto_library) which provides
-        the compiled code of any message that the services depend on.
-      well_known_protos: Should this library additionally depend on well known
-        protos
-      use_external: When True the grpc deps are prefixed with //external. This
-        allows grpc to be used as a dependency in other bazel projects.
-      generate_mocks: When True, Google Mock code for client stub is generated.
-      **kwargs: rest of arguments, e.g., compatible_with and visibility.
-  """
-  if len(srcs) > 1:
-    fail("Only one srcs value supported", "srcs")
+    Assumes the generated classes will be used in cc_api_version = 2.
 
-  proto_target = "_" + name + "_only"
-  codegen_target = "_" + name + "_codegen"
-  codegen_grpc_target = "_" + name + "_grpc_codegen"
-  proto_deps = ["_" + dep + "_only" for dep in deps if dep.find(':') == -1]
-  proto_deps += [dep.split(':')[0] + ':' + "_" + dep.split(':')[1] + "_only" for dep in deps if dep.find(':') != -1]
+    Args:
+        name (str): Name of rule.
+        srcs (list): A single .proto file which contains services definitions,
```

Comment:
Adding type annotations in the docstring is definitely an improvement, but it seems that this is setting a precedent. Where possible, I think it's a good idea to follow PyType conventions. So, how do you feel about changing `list` to `List[str]`?

Reply:
I don't mind doing it, but in Starlark case it is going to be either much harder or much less accurate than if it was a standard Python. For example, for `src` argument, even though we  specify strings, the actual type of that is `Label`, not `str`. Since it is getting complicated, I though that providing some minimal but at least correct information about the type is the best thing we can do, which still provides value and is not wrong/misleading or hard to maintain. 

WDYT?

Reply:
Fair point. Bazel itself appears to be [somewhat wishy-washy on the exact types](https://docs.bazel.build/versions/master/be/c-cpp.html#cc_binary_args). (Are they using ad-hoc polymorphism? I can't seem to construct a `Label` object instead of a string and pass it in.). It's fine as is.